lib/helpers/classic-invocation-component.js

```diff
@@ -0,0 +1,37 @@
+function transformTagName(tagName) {
+  if (isNestedComponentTagName(tagName)) {
+    return transformNestedTagName(tagName);
+  }
+
+  return capitalizedTagName(tagName);
+}
+
+function isComponentTagName(tagName) {
+  let hyphenatedComponentName = /\w+-\w+/g;
+  return hyphenatedComponentName.test(tagName);
+}
+
+function isNestedComponentTagName(tagName) {
+  let nestedComponentName = /\//g;
+  return nestedComponentName.test(tagName);
+}
+
+function transformNestedTagName(tagName) {
+  let paths = tagName.split('/');
+  return paths.map(name => capitalizedTagName(name)).join('::');
+}
+
+function capitalizedTagName(tagname) {
```

Comment:
Can you add some tests for this function? It doesn't quite match [what Ember does](https://github.com/emberjs/ember.js/blob/b31998b6a0cccd22a8fb6fab21d24e5e7f2cb70d/packages/ember-template-compiler/lib/system/dasherize-component-name.ts), so I'd like to make sure that at least [Ember's test input/outputs match](https://github.com/emberjs/ember.js/blob/b31998b6a0cccd22a8fb6fab21d24e5e7f2cb70d/packages/ember-template-compiler/tests/system/dasherize-component-name-test.js#L8-L24).